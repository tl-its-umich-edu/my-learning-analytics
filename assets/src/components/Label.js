import React from 'react'

function Label (props) {
  const {
    left,
    color,
    labelPlacement,
    labelText
  } = props

  return (
    <div
      style={{
        position: 'absolute',
        display: 'inline-block',
        color,
        left: labelPlacement === 'down'
          ? `calc(${left} - 115px)`
          : left,
        zIndex: 1,
        top: labelPlacement === 'down'
          ? '55px'
          : labelPlacement === 'downLower'
            ? '75px'
            : labelPlacement === 'up'
              ? '-25px'
              : 0
      }}
    >
      {labelText}
    </div>
  )
}

export default Label
