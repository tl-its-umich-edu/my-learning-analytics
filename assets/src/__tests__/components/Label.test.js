/* global it, expect */

import React from 'react'
import Label from '../../components/Label'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Label
        left={5}
        color='blue'
        labelPlacement='up'
        labelText='Hello world'
      />
    ).toJSON()

  expect(tree).toMatchSnapshot()
})
