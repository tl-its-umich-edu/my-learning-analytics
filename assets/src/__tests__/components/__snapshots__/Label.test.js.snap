// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  style={
    Object {
      "color": "blue",
      "display": "inline-block",
      "left": 5,
      "position": "absolute",
      "top": "-25px",
      "zIndex": 1,
    }
  }
>
  Hello world
</div>
`;
